### PR TITLE
Separate contracts source into dedicated files

### PR DESCRIPTION
**Pull request by 3esmit**
Fixes vacp2p/minime#15

## Description

Put one contract per file.

Note: no changes in gas report.

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Added natspec comments?
- [x] Ran `forge snapshot`?
- [x] Ran `pnpm lint`?
- [x] Ran `forge test`?

### CODE REVIEW

File: contracts/ApproveAndCallFallBack.sol [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+/// @title Approve And Call Fallback Token Interface
```
Comment (0x-r4bbit) [inline contracts/ApproveAndCallFallBack.sol:4]: Shouldn't this be `@notice`?

File: contracts/sample/Owned.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        owner = msg.sender;
+    }
+
+    /// @notice `owner` can step down and assign some other address to this role
+    /// @param _newOwner The address of the new owner. 0x0 can be used to create
+    ///  an ublock.timestampned neutral vault, however that cannot be undone
+    function changeOwner(address _newOwner) public onlyOwner {
+        owner = _newOwner;
+    }
+}
```
Comment (0x-r4bbit) [inline contracts/sample/Owned.sol:29]: We **could** think about making use of OZs `Ownable|2Step` but we can keep that in mind for later

**Review by 0x-r4bbit — APPROVED** [review]
Great job @3esmit!
Nice and clean.

Just a tiny comment.

File: contracts/ApproveAndCallFallBack.sol (hunk 1 above)
Comment (3esmit) [reply contracts/ApproveAndCallFallBack.sol:4]: Seems like it should be `@dev` "Explain to a developer any extra details"
`@notice` is for "Explain to an end user what this does"
https://docs.soliditylang.org/en/v0.8.21/natspec-format.html#tags

File: contracts/sample/Owned.sol (hunk 2 above)
Comment (3esmit) [reply contracts/sample/Owned.sol:29]: We probably use just Ownable, but we might just drop these contracts later.

**Pull request merged**